src/main/java/com/ning/http/client/providers/grizzly/AsyncHttpClientFilter.java

```diff
@@ -350,9 +350,10 @@ private void setProxyAuthorizationHeader(final Request req,
         }
         
         if (!isUsedConnection) {
+            boolean properProxyAuthorization = config.getProperProxyAuthorization();
             final String conAuth =
                     AuthenticatorUtils.perConnectionProxyAuthorizationHeader(
-                            req, proxy, isConnect);
+                            req, proxy, isConnect, properProxyAuthorization);
```

Comment:
Do not break the methods contract, simply use a system property in the class where the behavior is changed (AuthenticatorUtils). This is only a fail safe sysprop.

Reply:
Done.